model/src/main/java/org/gorpipe/gor/driver/PluggableGorDriver.java

```diff
@@ -45,15 +45,19 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.*;
-import java.util.Map.Entry;
 
 import static org.gorpipe.gor.driver.meta.DataType.LINK;
 
 public class PluggableGorDriver implements GorDriver {
     private final static Logger log = LoggerFactory.getLogger(PluggableGorDriver.class);
 
-    private final TreeMap<String, SourceType> protocolToSourceType = new TreeMap<>();
-    private final Map<SourceType, SourceProvider> sourceTypeToSourceProvider = new HashMap<>();
+    private final TreeMap<SourceType, SourceProvider> sourceTypeToSourceProvider = new TreeMap<>(new Comparator<SourceType>() {
+        @Override
+        public int compare(SourceType o1, SourceType o2) {
```

Comment:
Is the priority used to distinguish between general http urls and OCI http urls?

Reply:
Kind of, letting the OCI get the first change of picking them up.

Reply:
So the highest priority should be on those that are very specific, then generic ones, files (that picks up the rest) should have the lowest.